assets/javascripts/needleeditor.js

```diff
@@ -463,6 +500,17 @@ function saveNeedle(e) {
       addFlash('danger', '<strong>Unable to save needle:</strong><ul><li>' + errors.join('</li><li>') + '</li></ul>');
       return false;
   }
+
+  if ($("#property_workaround").prop("checked") && $("#input_workaround_desc").val() == "") {
```

Comment:
Unless you require the implicit conversion use `===` and `!==` in JavaScript. But this is likely better written as `!$("#input_workaround_desc").val()`.

Reply:
I will fix it